File: find-the-array-concatenation-value.ts
{
    function findTheArrayConcVal(nums: number[]): number {
        let c = 0;

        while(nums.length) {
            if(nums.length !== 1) {
                c += parseInt(`${nums.shift()}${nums.pop()}`);
            } else {
                c += nums.shift() || 0;
            }
        }

        return c;
    };

    console.log(findTheArrayConcVal([7,52,2,4]))
    console.log(findTheArrayConcVal([5,14,13,8,12]))
}

File: find-the-array-concatenation-value.rs
pub fn find_the_array_conc_val(nums: Vec<i32>) -> i64 {
    let mut c = 0;

    let mut new_array = nums;

    while !new_array.is_empty() {
            let first = new_array.remove(0);
            
            let last = if let Some(last) = new_array.pop() {
                last.to_string()
            } else {
                "".to_string()
            };

            let char = format!("{}{}", first , last);

            let num: Result<i64, _> = char.parse();

            match num {
                Ok(number) => c += number,
                Err(e) => println!("{}", e),
            }; 
    }

    c
}